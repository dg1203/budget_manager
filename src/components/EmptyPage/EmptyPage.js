import React from "react";
import { Button, Typography } from "@material-ui/core";
import PlusOneIcon from "@material-ui/icons/PlusOne";
import { makeStyles } from "@material-ui/core/styles";

const useStyles = makeStyles({
  root: {
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
    flexDirection: "column",
    padding: "25px",
    position: "absolute",
    margin: "auto",
    left: 0,
    right: 0,
    top: 0,
    bottom: 0,
    zIndex: -1
  }
});

const EmptyPage = () => {
  const classes = useStyles();
  return (
    <div className={classes.root}>
      <Typography gutterBottom variant="h5" component="h2">
        Lista jest pusta. Stwórz swój pierwszy słoik.
      </Typography>
      <Button variant="contained" color="primary" endIcon={<PlusOneIcon />}>
        Dodaj
      </Button>
    </div>
  );
};

export default EmptyPage;
