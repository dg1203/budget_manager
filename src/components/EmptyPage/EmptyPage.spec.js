import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router";
import EmptyPage from "./EmptyPage";
import store from "../../store";

const renderComponent = () =>
  render(
    <Provider store={store}>
      <MemoryRouter>
        <EmptyPage />
      </MemoryRouter>
    </Provider>
  );

describe("EmptyPage component", () => {
  it("renders the component", () => {
    const container = renderComponent();
    expect(container).toMatchSnapshot();
  });
});
