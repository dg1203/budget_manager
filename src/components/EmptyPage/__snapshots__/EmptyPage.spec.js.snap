// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyPage component renders the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="makeStyles-root-1"
      >
        <h2
          class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
        >
          Lista jest pusta. Stwórz swój pierwszy słoik.
        </h2>
        <a
          href="/jar/add"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Dodaj
              <span
                class="MuiButton-endIcon MuiButton-iconSizeMedium"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M10 8H8v4H4v2h4v4h2v-4h4v-2h-4zm4.5-1.92V7.9l2.5-.5V18h2V5z"
                  />
                </svg>
              </span>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="makeStyles-root-1"
    >
      <h2
        class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
      >
        Lista jest pusta. Stwórz swój pierwszy słoik.
      </h2>
      <a
        href="/jar/add"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Dodaj
            <span
              class="MuiButton-endIcon MuiButton-iconSizeMedium"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 8H8v4H4v2h4v4h2v-4h4v-2h-4zm4.5-1.92V7.9l2.5-.5V18h2V5z"
                />
              </svg>
            </span>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
